### DISCUSSION TITLE
Consider user storage api instead of localstorage

### DISCUSSION THREAD

**Discussion by MikeZeDev** [night/betterttv#6176] [Ideas]
Hello, my Firefox is set up to clean cache cookies and stuff after closing. Doing that clear BTTV settings.
I think setting should be preserved, like for any other addons.

I know you use local storage to store the settings. I know about https://github.com/night/betterttv/issues/5707 and the user was kinda rude.I also whitelist twitch domains but it makes me stay logged so wiping data at closing lose its purpose.

I have like a crapload of addons, with their own settings. Whatever id do (like using Bleachbit to purge some stuff) i never ever lose any setting. BTTV is the only one with that problem.

Any technical reasons to still use local storage? Would you consider a PR to switch (i dont know anything about extension coding but i could try, maybe)? Its even stated in Firefox documentation that localstorage shouldnt be used to store extension data, because clearing localstorage is an expected behavior : https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage

    Although this API is similar to [Window.localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) it is recommended that you don't use Window.localStorage in the extension code to store extension-related data. Firefox will clear data stored by extensions using the localStorage API in various scenarios where users clear their browsing history and data for privacy reasons, while data saved using the [storage.local](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/local) API will be correctly persisted in these scenarios.

Also, as stated in the Cloud storage topic, it could greatly help because storage api content can be synced between browser instance in Firefox and Chrome.

Thank for this nice addon anyway.

**Comment by pbaykalov** [comment]
Honestly this is just a fringe use case. 99,99% of users do not have a reason to clear cookies on exit.
I'd suggest that you just make a userscript for Tampermonkey which saves and restores local storage as you wish.

**Reply by ElTipejoLoco** [reply]
With the GraphQL error making the rounds on Twitch, clearing cache aggressively and frequently until it's fixed does make BTTV as an extension having its own storage more attractive, at least.

**Comment by dclstn** [comment]
Not a direct fix but, we've added a new feature "Cloud Backups" which will sync setting state between sessions that have it enabled. You can find the setting in the BetterTTV Settings Menu -> User Settings -> Enable Cloud Backups.

Though this feature is only available to users subscribed to BetterTTV Pro.
